src/heartbeats.cpp

```diff
@@ -494,52 +574,65 @@ void MegaBackupMonitor::onSyncStateChanged(MegaApi* /*api*/, MegaSync *sync)
     updateOrRegisterSync(sync);
 }
 
-std::shared_ptr<HeartBeatSyncInfo> MegaBackupMonitor::getHeartBeatBackupInfoByTransfer(MegaTransfer *transfer)
+void MegaBackupMonitor::onSyncDeleted(MegaApi *api, MegaSync *sync)
 {
-    if (!transfer->isSyncTransfer())
+    auto hBPair = mHeartBeatedSyncs.find(sync->getTag());
+    if (hBPair != mHeartBeatedSyncs.end())
     {
-        return nullptr;
-    }
-
-    int syncTag = 0;
+        mClient->reqs.add(new CommandBackupRemove(mClient, hBPair->second->backupId()));
 
-    // use map to get the syncTag directly if there was one
-    auto mTSPair = mTransferToSyncMap.find(transfer->getTag());
-    if (mTSPair != mTransferToSyncMap.end())
-    {
-        syncTag = mTSPair->second;
+        mHeartBeatedSyncs.erase(hBPair); //This is speculative: could be moved to backupremove_result
+        // in case we wanted to handle possible failing cases.
+        mPendingSyncUpdates.erase(sync->getTag()); // remove any pending update: also speculative.
     }
+}
+#endif
 
-    if (!syncTag) //first time
+std::shared_ptr<HeartBeatTransferProgressedInfo> MegaBackupMonitor::getHeartBeatBackupInfoByTransfer(MegaTransfer *transfer)
+{
+#ifdef ENABLE_SYNC
+    if (transfer->isSyncTransfer())
     {
-        Node *n = mClient->nodebyhandle(transfer->getType() == MegaTransfer::TYPE_UPLOAD ? transfer->getParentHandle() : transfer->getNodeHandle());
-        while (n)
+        int syncTag = 0;
+
+        // use map to get the syncTag directly if there was one
+        auto mTSPair = mTransferToSyncMap.find(transfer->getTag());
+        if (mTSPair != mTransferToSyncMap.end())
         {
-            if (n && n->localnode && n->localnode->sync)
-            {
-                syncTag = n->localnode->sync->tag;
-                mTransferToSyncMap[transfer->getTag()] = syncTag;
-                break;
-            }
-            LOG_warn << "Heartbeat could not get sync tag direclty from transfer handle. Going up";
-            n = n->parent;
+            syncTag = mTSPair->second;
         }
-    }
 
-    if (syncTag)
-    {
-        auto hBPair = mHeartBeatedSyncs.find(syncTag);
-        if (hBPair != mHeartBeatedSyncs.end()) //transfer info arrived before having created the HeartBeatSyncInfo object
+        if (!syncTag) //first time
         {
-            return hBPair->second;
+            Node *n = mClient->nodebyhandle(transfer->getType() == MegaTransfer::TYPE_UPLOAD ? transfer->getParentHandle() : transfer->getNodeHandle());
+            while (n)
+            {
+                if (n && n->localnode && n->localnode->sync)
```

Comment:
```suggestion
                if (n->localnode && n->localnode->sync)
```
`n` is already checked in the `while()`